Core/Src/throttle.c

```diff
@@ -71,32 +50,59 @@ void throttle_read(void *argument) {
 
 		log_data(ID_BRAKE, brake_status);
 
-		if (	APPS2 < 260
-			||	APPS2 >= 3720
-			|| 	APPS1 < 1802.24
-			|| 	APPS1 > 3900
-			|| 	APPS1 < apps1_calc * (1-APPS_PLAUSIBILITY_PERCENTAGE_TOLERANCE/100.0)
-			|| 	APPS1 > apps1_calc * (1+APPS_PLAUSIBILITY_PERCENTAGE_TOLERANCE/100.0))
+
+		//verifica a plausabilidade dos APPSs
+		if (check_APPS_error(APPS1, APPS2, aux_throttle_percent)){
+			aux_throttle_percent = 0;									//zera a a porcentagem do pedal
 			osEventFlagsSet(ECU_control_event_id, APPS_ERROR_FLAG); 	//caso tenha algum erro de plausabilidade do apps a flag será setada
+		}
 		else
 			osEventFlagsClear(ECU_control_event_id, APPS_ERROR_FLAG); 	//limpa a flag se n houver erro
 
+		//verifica a plausabilidade do BSE
 		if (aux_throttle_percent > 300 && BSE > 2200) {
-			aux_throttle_percent = 0;
+			aux_throttle_percent = 0;								//zera a a porcentagem do pedal
 			osEventFlagsSet(ECU_control_event_id, BSE_ERROR_FLAG);	//caso tenha algum erro de plausabilidade do BSE a flag será setada
 		}
 		else
-			osEventFlagsClear(ECU_control_event_id, BSE_ERROR_FLAG);//limpa a flag se n houver erro
+			osEventFlagsClear(ECU_control_event_id, BSE_ERROR_FLAG); //limpa a flag se n houver erro
+
 
 		throttle_percent = aux_throttle_percent;
 
 		log_data(ID_THROTTLE, throttle_percent);
 
-		//TODO: Averiguar duracao do delay
-		osDelay(25);
-
+		osDelay(THROTTLE_DELAY);
 	}
+}
 
-//	return (acelerador);
+bool check_APPS_error (uint16_t APPS1, uint16_t APPS2, uint16_t aux_throttle_percent){
+	uint16_t apps1_calc = 0;
+
+	//calcula o valor teórico de APPS1 a partir do valor de APPS2
+	if (aux_throttle_percent >= 0 && aux_throttle_percent < 200)
+		apps1_calc = 2212;
+	else if (aux_throttle_percent >= 200 && aux_throttle_percent < 400)
+		apps1_calc = (uint16_t) (1.679 * aux_throttle_percent + 1876);
+	else if (aux_throttle_percent >= 400 && aux_throttle_percent < 600)
+		apps1_calc = (uint16_t) (2.621 * aux_throttle_percent + 1499);
+	else if (aux_throttle_percent >= 600 && aux_throttle_percent < 800)
+		apps1_calc = (uint16_t) (2.212 * aux_throttle_percent + 1745);
+	else if (aux_throttle_percent >= 800 && aux_throttle_percent < 1135)
+		apps1_calc = (uint16_t) (1.515 * aux_throttle_percent + 2302);
+
+
+	if (APPS2 >= 3720) 			//Se o valor de APPS2 for acima do seu máximo
+		return true;
+	else if (APPS1 < 1802.24)	//Se o valor de APPS1 for abaixo do seu mínimo
+		return true;
+	else if (APPS1 > 3900)		//Se o valor de APPS1 for acima do seu máximo
+		return true;
+	else if (APPS1 < apps1_calc * (1-APPS_PLAUSIBILITY_PERCENTAGE_TOLERANCE/100.0)) //verifica se APPS1 está abaixo do valor teórico de APPS1, considerando a tolerância
+		return true;
+	else if (APPS1 > apps1_calc * (1+APPS_PLAUSIBILITY_PERCENTAGE_TOLERANCE/100.0)) //verifica se APPS1 está acima do valor teórico de APPS1, considerando a tolerância
+		return true;
```

Comment:
pode deixar no estilo do || que tava antes, só deixa o comentário no lado de cada linha
```suggestion
	if (APPS2 >= 3720             /*Se o valor de APPS2 for acima do seu máximo*/
	     || APPS1 < 1802.24	    /*Se o valor de APPS1 for abaixo do seu mínimo*/
	     || APPS1 > 3900  	    /*Se o valor de APPS1 for acima do seu máximo*/
	     || APPS1 < apps1_calc * (1-APPS_PLAUSIBILITY_PERCENTAGE_TOLERANCE/100.0)  /*verifica se APPS1 está abaixo do valor teórico de APPS1, considerando a tolerância*/
	     || APPS1 > apps1_calc * (1+APPS_PLAUSIBILITY_PERCENTAGE_TOLERANCE/100.0)) /*verifica se APPS1 está acima do valor teórico de APPS1, considerando a tolerância*/
		return true;
```
aqui na interface do GitHub é mt difícil escrever isso indentado, mas faz bonitinho aí rsrs